### DISCUSSION TITLE
Csla 4.x and DataPortal_Create() method

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#2997] [General]
This question is related to #2994 

If we override  DataPortal_Create as
  ```cs
      protected override void DataPortal_Create()
  ```
we can't pass a parameter but if we do 

  ```cs
         private void DataPortal_Create(int myID)
   ```
then we are fine. Are we looking at any "side effects"? we will call Base.DataPortal_Create in 

  ```cs
         private void DataPortal_Create(int myID)
  ```

Regards

**Comment by rockfordlhotka** [comment]
That is a supported concept, so you should be fine.

**Reply by Chicagoan2016** [reply]
Thank you Sir, we have a strange use case(s) going here : ), we try to design our objects based on behavior (we learned it from your books over the years : )). 
We have a root object with several 'child' objects. The object graph is working fine, there is one edge case where one of the child object's data might not be in the database and the root object needs the primary key from that data.
I created a command object to see if that particular object's data exist in the database and then pass the primary key to the root object's DataPortal_Create and based on that I call either DataPortal.CreateChild or DataPortal.Fetchchild
but for some reason I am not a fan of this design! it assumes the UI developer will know to use the command object first.
